packages/select/src/withDelete.ts

```diff
@@ -1,77 +1,49 @@
 import {
-  deleteForward,
-    getNodeEntries,
-    getPointBefore,
-    isBlockAboveEmpty,
-    isSelectionExpanded,
-    PlateEditor,
-    queryNode,
-    removeNodes,
-    Value,
-    WithPlatePlugin,
-  } from '@udecode/plate-common';
-  
-  import { DeletePlugin } from './createDeletePlugin';
-import Slate from 'slate';
-  
-  /**
-   * Set a list of element types to select on backspace
-   */
-  export const withDelete = <
-    V extends Value = Value,
-    E extends PlateEditor<V> = PlateEditor<V>,
-  >(
-    editor: E,
-    {
-      options: { query },
-    }: WithPlatePlugin<DeletePlugin, V, E>
-  ) => {
-    const { deleteForward } = editor;
-  editor.deleteForward = (unit: 'character' | 'word' | 'line' | 'block') => {
-    const { selection } = editor; 
-    console.log("outer" ,query);
-    if(!editor.selection) return;
-    
+  getAboveNode,
+  isBlockAboveEmpty,
+  isSelectionExpanded,
+  PlateEditor,
+  queryNode,
+  removeNodes,
+  Value,
+  WithPlatePlugin,
+} from '@udecode/plate-common';
+
+import { DeletePlugin } from './createDeletePlugin';
 
+/**
+ * Set a list of element types to select on backspace
+ */
+export const withDelete = <
+  V extends Value = Value,
+  E extends PlateEditor<V> = PlateEditor<V>,
+>(
+  editor: E,
+  { options: { query } }: WithPlatePlugin<DeletePlugin, V, E>
+) => {
+  const { deleteForward } = editor;
+  editor.deleteForward = (unit: 'character' | 'word' | 'line' | 'block') => {
+    if (!editor.selection) return;
     if (!isSelectionExpanded(editor) && isBlockAboveEmpty(editor)) {
-      console.log("asdas");
+      // check when line is empty
+      const isValidNode = queryNode(getAboveNode(editor), query);
       if (query) {
```

Comment:
The logic looks better now, but there's still a lot of unnecessary duplication of the `deleteForward` and `removeNodes` lines, which makes the code harder to read.

There are only two cases: (1) it calls `deleteForward`, or (2) it calls `removeNodes`. You should be able to structure the code to use a single if statement (discounting the `!editor.selection` line at the top).

Reply:
code has been updated